src/test/scala/vsys/blockchain/contract/DataEntrySpecification.scala

```diff
@@ -40,19 +73,88 @@ class DataEntrySpecification extends PropSpec with PropertyChecks with Generator
 
       assertEys(recovered, pk)
     }
+
+    forAll(validInt32EntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
+    forAll(validShortTextEntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
+    forAll(validContractAccountEntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
+    forAll(validTimestampEntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
+    forAll(validBooleanEntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
+    forAll(validShortBytesEntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
+    forAll(validOpcBlockEntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
+    forAll(validBigIntegerEntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
   }
 
-  property("report invalid data type") {
+  property("report invalid and valid data type") {
     val byteArray1 = Array[Byte](0, 118, 97, 108, 117, 101, 49)
     val byteArray2 = Array[Byte](4, 118, 97, 108, 117, 101, 49)
     val byteArray3 = Array.fill[Byte](1 + 32)(1)
     val byteArray4 = Array.fill[Byte](1 + 26)(2)
     val byteArray5 = Array.fill[Byte](1 + 8)(3)
+    val byteArray6 = Array.fill[Byte](1 + 4)(4)
+    val byteArray7 = Array[Byte](5, 0, 0)
+    val byteArray8 = Array.fill[Byte](1 + 26)(6)
+    val byteArray9 = Array.fill[Byte](1 + 26)(7)
+    val byteArray10 = Array.fill[Byte](1 + 8)(9)
+    val byteArray11 = Array[Byte](10, 1)
+    val byteArray12 = Array[Byte](11, 0, 0)
+    val byteArray13 = Array[Byte](12, 118, 97, 108, 117, 101, 49)
+    val byteArray14 = Array[Byte](13, 0, 0)
+    val byteArray15 = Array[Byte](14, 0, 0)
```

Comment:
just confused the representation of empty byte array in BigInteger~ @zl730

Reply:
`java.lang.NumberFormatException: Zero length BigInteger` would be caused